src/riscv-unpriv-integration.adoc

```diff
@@ -0,0 +1,388 @@
+For instructions that have differing behavior in {cheri_cap_mode_name} and {cheri_int_mode_name} (memory access and pointer arithmetic instructions), the descriptions will list assembler syntax and behavior for both modes.
+
+==== Capability checks
+
+Every memory access performed by a CHERI core must be authorized by a capability.
+The authorizing capability is either named explicitly (the base register of a load/store operation) or implicitly (when executing a branch, <<pcc>> is used for authorization).
+For example, `lw t0, 16(csp)` loads a word from memory, getting the address, bounds, and permissions from the `csp` register.
+
+Instruction fetch is also authorized by a capability: the program counter capability (<<pcc>>) which extends PC.
+This allows code fetch to be bounded, preventing a wide range of attacks that subvert control flow with integer data.
+
+With {cheri_base_ext_name} instruction fetches, jumps, and memory accesses may result in a fatal trap if the authorizing capability does grant the required permissions or the access is out of bounds.
+No other exception paths are added: in particular, capability manipulations do not trap, but may clear the tag on the result capability if the operation is not permitted.
+
+[#pcc,reftext="pcc"]
+==== PCC - The Program Counter Capability
+
+An authorizing capability with appropriate permissions is required to execute
+instructions in {cheri_base_ext_name}. Therefore, the unprivileged program
+counter (*pc*) register is extended so that it is able to hold a capability.
+The extended register is called the program counter capability (<<pcc>>). The
+<<pcc>> address field is effectively the *pc* in the base RISC-V ISA so that the
+hardware automatically increments as instructions are executed. The <<pcc>>'s
+metadata and tag are reset to the <<infinite-cap>> capability metadata and tag
+with the address field set to the core boot address.
+
+The hardware performs the following checks on <<pcc>> for each instruction
+executed in addition to the checks already required by the base RISC-V ISA. A
+failing check causes a CHERI exception.
+
+* The tag must be set
+* The capability must not be sealed
+* The capability must grant execute permission
+* All bytes of the instruction must be in bounds
+
+NOTE: Operations that update <<pcc>>, such as changing privilege or executing
+jump instructions, unseal capabilities prior to writing. Therefore,
+implementations do not need to check that that <<pcc>> is unsealed when
+executing each instruction. However, this property has not yet been formally verified and may not hold if additional CHERI extensions beyond {cheri_base_ext_name} are implemented.
+
+.Program Counter Capability
+[#pcc-format]
+include::img/pccreg.edn[]
+
+<<pcc>> is a code capability, so it does not need to be able to hold all possible invalid addresses (see <<section_invalid_addr_conv>>).
```

Comment:
The xref says "invalid address conversion" but I think it should say "CHERI invalid address conversion"